### PR TITLE
Add npoints generator

### PR DESCRIPTION
**Pull request by jiayu-google**
Number of test points should increase as number of publishers increases. Added arguments to reflect this.

### CODE REVIEW

File: src/driver/latin_hypercube_random_test_point_generator.py [hunk 1]
```diff
@@ -48,11 +50,19 @@ def __init__(
           dataset:  The DataSet for which test points are to be generated.
           rng:  A numpy Generator object that is used to seed the generation
             of random test points.
+          npoints_generator: a function from the number of publishers to the
```
Comment (matthewclegg) [inline src/driver/latin_hypercube_random_test_point_generator.py:53]: Could we change the comment to read, 'A function that returns the number of test points that should be generated.  The argument of the function is the number of publishers.'

File: src/driver/latin_hypercube_random_test_point_generator.py [hunk 2]
```diff
@@ -48,11 +50,19 @@ def __init__(
...
+            number of test points. EΩxample is
```
Comment (matthewclegg) [inline src/driver/latin_hypercube_random_test_point_generator.py:54]: Fix typo in the word 'example'.

**Review by matthewclegg — APPROVED** [review]
This seems like a useful change.  Thanks for adding it.  One thing I wonder about is whether it makes sense to get rid of the 'npoints' argument, and instead set the default value for npoints_generator to lambda p: MINIMUM_NUMBER_OF_TEST_POINTS.  I am OK with the PR in its current form, but I thought you might want to consider this.

File: src/driver/uniformly_random_test_point_generator.py [hunk 3]
```diff
@@ -37,19 +38,28 @@ def __init__(
         self,
         dataset: DataSet,
         rng: np.random.Generator,
+        npoints_generator: Callable[[int], int] = lambda x: None,
         npoints: int = MINIMUM_NUMBER_OF_TEST_POINTS,
     ):
-        """Returns a UniformlyRandomTestPointGenerator.
+        """Returns a LatinHypercubeRandomTestPointGenerator.
```
Comment (riemanli) [inline src/driver/uniformly_random_test_point_generator.py:44]: I might misunderstand something, but I feel the original is correct 😅

Comment (jiayu-google) [reply src/driver/uniformly_random_test_point_generator.py:44]: Thanks for catching this! I did too many copy pastes 😅

File: src/driver/latin_hypercube_random_test_point_generator.py (hunk 2 above)
Comment (jiayu-google) [reply src/driver/latin_hypercube_random_test_point_generator.py:54]: Thanks for catching this!

File: src/driver/latin_hypercube_random_test_point_generator.py (hunk 1 above)
Comment (jiayu-google) [reply src/driver/latin_hypercube_random_test_point_generator.py:53]: Sure. It does read better.

**Review by riemanli — APPROVED** [review]

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**